nodebuilder/header/module.go

```diff
@@ -32,6 +32,9 @@ func ConstructModule[H libhead.Header[H]](tp node.Type, cfg *Config) fx.Option {
 		fx.Provide(newHeaderService),
 		fx.Provide(newInitStore[H]),
 		fx.Provide(func(subscriber *p2p.Subscriber[H]) libhead.Subscriber[H] {
+			if subscriber == nil {
+				return nil
+			}
```

Comment:
instead of short circuits approach, do not provide unwanted componentets to the fx.